### DISCUSSION TITLE
Command for jumping to marker file?

### DISCUSSION THREAD

**Discussion by shouya** [bbatsov/projectile#1843] [Ideas]
It's quite an often need to jump to the project file of the current project. By project file I mean the file type like `Cargo.toml` for Rust project, `Gemfile` for Ruby, `mix.exs` for Elixir, etc. Incidentally, this kinds of files are usually what's used as marker by `projectile`.

In general, I wish there is a way to quickly jump to a marker file with one key stroke. The jump need not to be entirely accurate, but hey, `projectile-find-test-file` isn't accurate all the time either, so I can bare with that.

It's weird I never see any one mention about this feature from the issue list. I myself certainly needed this feature every 10-20 minutes or so when writing any code.

So, currently I simply hacked this command out from scratch. I'm sharing it because I suppose others may also have the same need:

```emacs-lisp
(defun shou/projectile-find-marker-file ()
  "Find the marker file (Cargo.toml, mix.exs, etc) for the current project."
  (interactive)
  (if-let* ((this-file (buffer-file-name))
            (marker-files (projectile-project-type-attribute
                           (projectile-project-type)
                           'marker-files))
            (get-target-marker-file
             (lambda (f)
               (when-let* ((dir (locate-dominating-file default-directory f))
                           (full-path (concat dir f))
                           (_ (not (f-same-p this-file full-path))))
                 full-path)))
            (marker-file (seq-find get-target-marker-file marker-files))
            (target-marker-file (funcall get-target-marker-file marker-file)))
      (find-file target-marker-file)
    (if-let* ((default-directory (f-dirname default-directory))
              (parent-marker-file (seq-find get-target-marker-file marker-files))
              (target-parent-marker-file (funcall get-target-marker-file parent-marker-file)))
        (find-file target-parent-marker-file)
      (message "No marker file found"))))

(with-eval-after-load 'projectile
  (bind-key "m" 'shou/projectile-find-marker-file projectile-command-map))
```

I could file a PR to add this command to projectile if @bbatsov deems this feature worthful.

**Comment by bbatsov** [comment]
Yeah, that'd be a welcome feature. For some reason I thought I had implemented it, but I guess my memory betrayed me here. 😅
